lib/yard-gobject-introspection.rb

```diff
@@ -5,8 +5,12 @@
 class GObjectIntropsectionHandler < YARD::Handlers::Ruby::Base
   handles :module
 
+  @@has_processed_module = false
+
   def process
-    gir_path = File.expand_path("gir-1.0", RbConfig::CONFIG["datadir"])
+    return if @@has_processed_module
+
+    gir_path = ENV['GIR_PATH'] || File.expand_path("gir-1.0", RbConfig::CONFIG["datadir"])
```

Comment:
How about `GI_REPOSITORY_PATH` instead of `GIR_PATH`?
Because other GObject Introspection tools use `GI_TYPELIB_PATH` for `.typelib`.

Reply:
How about `GI_GIR_PATH`? Seems to be used here: https://gitlab.gnome.org/GNOME/gobject-introspection/-/commit/3b68d55f1a9aa2a96ad6e51eaa8949ec744a91f7

Reply:
Let's use it!